### ISSUE TITLE
refactor!: define colors in tailwind config instead of hardcoding

### ISSUE THREAD

**Issue by hubertmalkowski** [mishka-group/mishka_chelekom#124]
#### Background
The library aims to be as general-purpose as possible. To support this goal, the current implementation needs improvement. At present, [colors are hardcoded in components](https://github.com/mishka-group/mishka_chelekom/blob/master/priv/templates/components/button.eex#L608), which limits the customizability of the library. This approach creates complexity for users who want to adapt the components to match their specific branding, forcing them to manually change colors across many many components.

#### Solution
Implement a color system by integrating color tokens directly into the Tailwind configuration. The color tokens should be generated and inserted into `tailwind.config.js` via cli.

#### Loose thoughts
- With growing number of components - the later the color system is implemented the harder it is to adopt it
- Color tokens named after it's role and not by the color name (eg. "primary" instead of "blue" or "background" instead of "white")

#### References
 - [shadcn/ui](https://ui.shadcn.com/docs/theming) - has color tokens that are injected by cli to tailwind config
	 - [salad_ui](https://github.com/bluzky/salad_ui) which is Liveview implementation
 - [Material Design 3](https://m3.material.io/styles/color/system/overview) - example of fairly simple, well defined and widely used color system
 - [Fluent ui 2](https://react.fluentui.dev/?path=/docs/theme-colors--docs), [Carbon Design system](https://carbondesignsystem.com/elements/color/tokens/) - examples of big and complex color system

**Comment by shahryarjb** [comment]
Hello dear @hubertmalkowski ,

Our initial approach was to ensure that users wouldn’t need to modify any files themselves. Additionally, each component would remain isolated and independent, with dependencies limited solely to our other components.

To achieve this, we hardcoded all the colors directly within each component. In this current phase (as you can see from the list of issues), we are re-implementing all the colors because of our redesign. Once this phase is complete, we’ll focus on improving our CLI to allow hardcoded colors to be converted into Tailwind configurations with a single command. However, I believe this will be part of phase 0.0.3, and we have plans for it.

Rest assured, we’ll definitely implement this, as it has also been suggested in the Elixir forum before. It’s worth noting that we’re not solely relying on this solution for color customization. In later phases, our CLI will generate configuration files, enabling developers to easily customize either individual components or the entire library to suit their needs.

Regarding naming conventions:
If you’ve noticed, our naming for colors and sizes is consistent. For example, the color named warning represents a shade of brown-orange. This naming convention helps users, especially those less experienced in front-end development, to avoid confusion with various token names for colors.

However, this is not a limitation. Advanced users or those familiar with the system can assign any custom color they prefer, as all properties support customization.

In the next phase, we’ll be adding a section to our documentation titled "Design System." This will include all named colors, token colors, and their definitions. Over time, we’ll expand on these core colors.

Thank you for the references you provided; we’ll definitely review them soon. It may take a little time as we transition to our new infrastructure.

We truly appreciate your suggestion and contributions to improving this project. We’re committed to simplifying everything further and ensuring that any manual operation can be replaced with an automated generator-based solution.

> If you have any other suggestions or if you encounter any issues, please feel free to let us know or open a new issue. We’ll be more than happy to assist you.

Thank you!

**Comment by kantum** [comment]
Beware `tailwind.config.js` file will be removed in V4 https://tailwindcss.com/docs/v4-beta#css-first-configuration

**Comment by MonaAghili** [comment]
Hi @kantum,

Thank you for bringing this to our attention. We're currently reviewing the new beta version of Tailwind CSS to ensure we make the most informed and best decision possible.
Best regards

**Comment by shahryarjb** [comment]
We closed this issue to re-search about new tailwind options and we have separated task for it. We will implement a good option for this lib.
thank you
